test/core/transport/promise_endpoint_test.cc

```diff
@@ -0,0 +1,1389 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/transport/promise_endpoint.h"
+
+// IWYU pragma: no_include <sys/socket.h>
+
+#include <algorithm>
+#include <cstring>
+#include <functional>
+#include <memory>
+#include <queue>
+#include <string>
+#include <tuple>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/types/optional.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/event_engine/port.h>  // IWYU pragma: keep
+#include <grpc/event_engine/slice_buffer.h>
+#include <grpc/support/log.h>
+
+#include "src/core/lib/promise/activity.h"
+#include "src/core/lib/promise/detail/basic_join.h"
+#include "src/core/lib/promise/join.h"
+#include "src/core/lib/promise/seq.h"
+#include "src/core/lib/slice/slice.h"
+#include "src/core/lib/slice/slice_buffer.h"
+#include "src/core/lib/slice/slice_internal.h"
+#include "test/core/promise/test_wakeup_schedulers.h"
+
+using testing::MockFunction;
+using testing::StrictMock;
+
+namespace grpc_core {
+namespace testing {
+
+class MockEndpoint
```

Comment:
It's unclear to me what we're gaining by this being a mock.

At present (though I could be convinced otherwise) I think we'd be better off having a `FakeEndpoint` - still deriving from `grpc_event_engine::experimental::EventEngine::Endpoint` but directly implementing Read, Write, etc.

The rest of the classes working would be the same, but we'd drop a bunch of the expect calls in tests and end up with a more readable thing.

Reply:
I see, by less expect calls you mean remove the ones that related to the `EventEngine::Endpoint` mocks or in general, the expect calls that are redundant?

Reply:
As discuss offline, I update the test with mock methods only, and remove the fake endpoint logics.